app/models/story_parser.rb

```diff
@@ -421,8 +421,12 @@ def parse_author_common(email, name)
       name = name.to_ascii.gsub(/[^\w[ \-@.]]/u, "")
       external_author = ExternalAuthor.find_or_create_by(email: email)
       external_author_name = external_author.default_name
+
+      # if the name and email don't exist in the DB tables, add it
       unless name.blank?
-        external_author_name = ExternalAuthorName.where(name: name, external_author_id: external_author.id).first ||
+        raise Error, external_author.errors.full_messages.join(" ") if external_author.invalid?
+
+        external_author_name = ExternalAuthorName.find_by(name: name, external_author_id: external_author.id) ||
```

Comment:
Does this comment look accurate? I didn't understand the intention of this part of the code first, and the comment I inserted was my best guess.